src/transformers/models/time_series_transformer/modeling_time_series_transformer.py

```diff
@@ -0,0 +1,1973 @@
+# coding=utf-8
+# Copyright 2022 kashif The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch TimeSeriesTransformer model. """
+
+import math
+import copy
+import random
+from typing import Optional, Tuple, List
+
+import torch
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from gluonts.torch.modules.scaler import MeanScaler, NOPScaler
+from gluonts.torch.modules.feature import FeatureEmbedder
+from gluonts.torch.distributions import StudentTOutput
+from gluonts.torch.modules.loss import NegativeLogLikelihood
+from gluonts.torch.util import weighted_average
```

Comment:
So as discussed, these should eventually all be implemented directly in this modeling file.

It remains to be seen whether we later on define them in a separate `time_series_utils.py` script, or instead use Copied from statements in case we're adding new time series models. Currently I'm in favor of the later (duplication rather than making the code more modular)